BedrockCore.cpp

```diff
@@ -64,6 +64,70 @@ bool BedrockCore::isTimedOut(unique_ptr<BedrockCommand>& command) {
     return false;
 }
 
+void BedrockCore::prePeekCommand(unique_ptr<BedrockCommand>& command) {
+    AutoTimer timer(command, BedrockCommand::PREPEEK);
+    BedrockServer::ScopedStateSnapshot snapshot(_server);
+    command->lastPeekedOrProcessedInState = _server.getState();
+
+    // Convenience references to commonly used properties.
+    const SData& request = command->request;
+    SData& response = command->response;
+    STable& content = command->jsonContent;
+
+    try {
+        SDEBUG("prePeeking at '" << request.methodLine << "' with priority: " << command->priority);
+        uint64_t timeout = _getRemainingTime(command, false);
+        command->prePeekCount++;
+
+        _db.startTiming(timeout);
+
+        if (!_db.beginTransaction(SQLite::TRANSACTION_TYPE::SHARED)) {
+            STHROW("501 Failed to begin shared prePeek transaction");
+        }
+
+        // Make sure no writes happen while in prePeek command
+        _db.setQueryOnly(true);
+
+        // prePeek.
+        command->reset(BedrockCommand::STAGE::PREPEEK);
+        command->prePeek(_db);
+        SDEBUG("Plugin '" << command->getName() << "' prePeeked command '" << request.methodLine << "'");
+
+        if (command->httpsRequests.size()) {
+            STHROW("405 https requests cannot be made in prePeek");
+        }
+
+        if (!content.empty()) {
```

Comment:
Why is this block here?

Reply:
The way I understand it, we do not actually return what is in `command->jsonResponse` to the client. Instead, we return what is in `command->response` to the client. This block makes sure that if something is added to `jsonResponse` in `prePeek` that needs to be returned to the client, that content actually gets returned to the client.

Reply:
Ah ok, this does some rewriting on the various types of responses we can create.